lib/algo/manifold_interp/SnapshotInterpolator.h

```diff
@@ -0,0 +1,51 @@
+#ifndef included_SnapshotInterpolator_h
+#define included_SnapshotInterpolator_h
+
+#include "Interpolator.h"
+#include <vector>
+#include <string>
+
+namespace CAROM {
+
+class Matrix;
+class Vector;
+
+/**
+ * Implements PCHIP algorithm.  Based on "A METHOD FOR CONSTRUCTING LOCAL MONOTONE
+ * PIECEWISE CUBIC INTERPOLANTS", Fritchs and Butland (1984).  as well as "MONOTONE
+ * PIECEWISE CUBIC INTERPOLATION," Fritsch and Carlson (1980)
+ *
+ */
+class SnapshotInterpolator
```

Comment:
`SnapshotInterpolator` sounds rather general. Would it be better to use a more specific name, like `PCHIPInterpolator`? That way, there will not be confusion with other interpolators added later.

Reply:
I'm open to changing the name if we get more time based interpolation schemes in the future.

Reply:
The problem with changing it later is that any code using this class has to be changed. A name involving PCHIP would also be more descriptive. What do other reviewers think? Does anyone have plans for future interpolators?

Reply:
I have changed the name to PCHIPInterpolator.  This was ultimately the correct decision.